src/parser/internal/utils.rs

```diff
@@ -180,25 +179,6 @@ pub fn braced<T>(
     })
 }
 
-/// Parse an item that is surrounded by brackets.
-///
-/// This function will skip the left bracket, call the given function,
-/// and then skip the right bracket.
-pub fn bracketed<T>(
-    state: &mut State,
-    func: &(dyn Fn(&mut State) -> ParseResult<T>),
-) -> ParseResult<Bracketed<T>> {
-    let left_bracket = skip_left_bracket(state)?;
-    let inner = func(state)?;
-    let right_bracket = skip_right_bracket(state)?;
-
-    Ok(Bracketed {
-        left_bracket,
-        inner,
-        right_bracket,
-    })
-}
-
```

Comment:
i believe i made a mistake by introducing these, the only one that should be kept is `CommaSeparated`.